py/objint.c

```diff
@@ -374,26 +374,27 @@ mp_obj_t mp_obj_int_binary_op_extra_cases(mp_uint_t op, mp_obj_t lhs_in, mp_obj_
 
 // this is a classmethod
 STATIC mp_obj_t int_from_bytes(size_t n_args, const mp_obj_t *args) {
-    // TODO: Support long ints
-    // TODO: Support byteorder param
+    // TODO: Better support byteorder param
     // TODO: Support signed param (assumes signed=False at the moment)
     (void)n_args;
 
-    if (args[2] != MP_OBJ_NEW_QSTR(MP_QSTR_little)) {
-        mp_not_implemented("");
-    }
-
     // get the buffer info
     mp_buffer_info_t bufinfo;
     mp_get_buffer_raise(args[1], &bufinfo, MP_BUFFER_READ);
 
-    // convert the bytes to an integer
+    #if MICROPY_LONGINT_IMPL != MICROPY_LONGINT_IMPL_NONE
```

Comment:
It would be desirable to use the original code path (ie convert to small-int) if the value fits in a small integer.  To keep it simple one could just do something like `if (MP_SMALL_INT_FITS(1 << (bufinfo.len * 8 -1))) ...`.